pkg/ottl/expression.go

```diff
@@ -143,6 +143,22 @@ func (l *listGetter[K]) Get(ctx context.Context, tCtx K) (any, error) {
 	return evaluated, nil
 }
 
+type mapGetter[K any] struct {
+	mapValues map[string]Getter[K]
+}
+
+func (m *mapGetter[K]) Get(ctx context.Context, tCtx K) (any, error) {
+	evaluated := map[string]any{}
+	for k, v := range m.mapValues {
+		val, err := v.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+		evaluated[k] = val
+	}
+	return evaluated, nil
```

Comment:
```suggestion
	return pcommon.NewMap().FromRaw(evaluated)
```
I prefer the ergonomics of Go maps, but the more I think about this, the more we should probably just standardize on `pcommon.Map` since that's already the map type used everywhere else.

Reply:
I looked into this, and I think we have to keep using `map[string]any` here: otherwise, it seems nested maps will not work, as when attempting to to invoke a pcommon.Map's `FromRaw()` method on a map containing a `pcommon.Map`, a error indicating that this is not possible will be returned: `<Invalid value type pcommon.Map>` 

However, the exported `Getters` will make sure that a caller will always receive a `pcommon.Map`, also when the internal `mapGetter` returns a `map[string]any` - e.g. here:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/expression.go#L366-L372

Reply:
Thanks for looking into it, I didn't realize `FromRaw` wouldn't take in other `pcommon.Map`s. Another option could be to wrap this Getter inside a `StandardPMapGetter` later, I'll leave another comment on how that could work.

I do see how a function that specifies an `ottl.PMapGetter` will have the map converted, but a function that only specifies an `ottl.Getter` will receive a `map[string]any`: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/functions.go#L424. Maybe it's not a huge deal, but it feels inconsistent.